src/App.tsx

```diff
@@ -58,11 +61,13 @@ type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
 type NumberMode = "normal" | "corner";
 
 interface BoardState {
-    highlights: Set<SquareAddress>;
+    continueHighlighting: boolean;
     contradicts: Set<SquareAddress>;
-    values: Map<SquareAddress, SquareValue>;
+    highlighting: boolean;
```

Comment:
It's not clear to me what the differences between `highlighting` and `continueHighlighting` are. Perhaps jsdocs will help but perhaps also consider renaming them to something that's unambiguous.